owncloud/richdocuments
# Migrate to oC Web v7

## Preface
ownCloud Web version 7.0.0 has been released just recently. It does not support Vue2 (EOL end of 2023) anymore as it moved on to Vue3. Furthermore there have been some improvements in the extension system for editor apps.

## Description
Making `richdocuments` compatible with ownCloud Web v7.

**Warning**: there was a fix necessary, so this only works with ownCloud Web v7.0.2 or higher.

## How to test as a dev
With the ownCloud Web dev stack this PR can be used for testing and development as follows:

1. clone the `richdocuments` repo and then check out this branch
2. check out the `stable-7.0` branch of ownCloud Web
3. adjust some files in oC Web:
- file `dev/docker/oc10.entrypoint.sh`, add `occ app:enable richdocuments` at the end of the file
- file `dev/docker/oc10.web.config.json`, add richdocuments as external app:
  ```json
  "external_apps": [
      {
        "id": "richdocuments",
        "path": "http://host.docker.internal:8080/index.php/apps/richdocuments/js/richdocuments.js"
      }
   ]
  ```
- file `docker-compose.yml`, add your local clone of this repo as volume to the oc10 container: `- /your/local/path/to/richdocuments:/mnt/data/apps/richdocuments`
4. `docker-compose up -d oc10`
5. `docker run -t -d -p 9980:9980 -e "extra_params=--o:ssl.enable=false" -e "username=admin" -e "password=admin" --name collabora --cap-add MKNOD collabora/code:6.4.8.6`
6. add `http://host.docker.internal:9980` as collabora url in the oc10 admin settings (last menu item `Additional`, then text field `Collabora Online server`)
7. switch to `New Design` in the ownCloud app switcher
8. try all the collabora features.
